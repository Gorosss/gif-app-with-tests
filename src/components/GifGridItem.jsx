import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {

    const [copied, setCopied] = useState(false);


    const copyToClipboard = (text) => {
        // Creamos un elemento textarea para copiar el texto
        const textarea = document.createElement('textarea');
        // Asignamos el texto que queremos copiar al valor del textarea
        textarea.value = text;
        // Hacemos que el textarea sea invisible
        textarea.style.position = 'fixed';
        textarea.style.top = '0';
        textarea.style.left = '0';
        textarea.style.opacity = '0';
        // AÃ±adimos el textarea al DOM
        document.body.appendChild(textarea);
        // Seleccionamos el texto dentro del textarea
        textarea.select();
        // Ejecutamos el comando de copiar
        document.execCommand('copy');
        // Removemos el textarea del DOM
        document.body.removeChild(textarea);

        setCopied(true);
        setTimeout(() => {
          setCopied(false);
        }, 2000);
      };

    const handleClickCard = (url) => {
        copyToClipboard(url)
    }

    return (
        <div className="card animate__animated animate__fadeIn" onClick={() => handleClickCard(url)}>
            {copied && <div style={{ position: 'fixed', transform: 'translateX(-50%)', background: 'rgba(0, 0, 0, 0.8)', color: '#fff', padding: '10px', borderRadius: '5px' }}>Copied!</div>}

            <img src={ url } alt={ title } />
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
