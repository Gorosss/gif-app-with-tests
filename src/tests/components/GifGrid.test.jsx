import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


jest.mock('../../hooks/useFetchGifs');


describe('Test <GifGrid />', () => {

    const category = 'One Punch';


    test('Render correctly', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('Have to show items on image load useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost',
            title: 'title'
        },
        {
            id: '123',
            url: 'https://localhost',
            title: 'title'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
        
        // expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    })
    
    

})

