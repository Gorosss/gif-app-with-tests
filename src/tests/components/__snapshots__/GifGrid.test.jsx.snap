// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGrid /> Render correctly 1`] = `ShallowWrapper {}`;
